### ISSUE TITLE
500 Server Error while creating task from cwl-tes

### ISSUE THREAD

**Issue by sgalpha01** [EMBL-EBI-TSI/tesk-api#42]
Python Version: `3.8.13`

I was using `cwl-tes` to submit a task to TESK. The task which I used can be found [here](https://github.com/uniqueg/cwl-example-workflows). The TESK api was deployed locally in `minikube` and was built using the [current source code](https://github.com/elixir-cloud-aai/tesk-api/commit/c4494e5066b176544e138023f5d04663ea7a5c3d) from the Github repository.

The error which is thrown in `cwl-tes` commandline is:
```
ERROR Workflow error:
500 Server Error:  for url: http://192.168.49.2:31567/v1/tasks
```
The URL refers to my local TESK deployment.

I'm sharing the exact error from the API side:
```
ERROR 1 --- [nio-8080-exec-1] u.a.e.t.tesk.tes.service.TesServiceImpl  : ERROR: In createTask
java.lang.IllegalArgumentException: No suffix for exponent-18
```
The entire logs can be found [here](https://pastebin.com/1HQB4wdj).

**Comment by lvarin** [comment]
The error seems to be here:

https://github.com/elixir-cloud-aai/tesk-api/blob/master/src/main/java/uk/ac/ebi/tsc/tesk/tes/service/TesServiceImpl.java#L51

**Comment by lvarin** [comment]
I will try to reproduce the error, can you paste here the exact command line you used and the yaml files you are using in the call?
That can save me a lot of time. :)

**Comment by sgalpha01** [comment]
Yeah sure :)
```bash
cwl-tes --debug --tes http://192.168.49.2:31567  --remote-storage-url ftp://ftp/home/tesk ../cwl-example-workflows/hashsplitter-workflow.cwl ../cwl-example-workflows/hashsplitter-test.yml
```
Check [this repo](https://github.com/uniqueg/cwl-example-workflows) for YAML files.

**Comment by lvarin** [comment]
I managed to reproduce the problem. I am still looking for the cause. For the moment I just know that the problem is while calling ".gson.toJson"

**Comment by lvarin** [comment]
The text of the exception comes from here:

https://github.com/kubernetes-client/java/blob/master/kubernetes/src/main/java/io/kubernetes/client/custom/SuffixFormatter.java#L111

**Comment by sgalpha01** [comment]
I'm not sure whatever I comment now would be of any help. I have attached a screenshot, please check that. It displays 6 commits. Except for the first one, after none of the commits I was able to package the api using `mvn package`. They were throwing some errors. The top one was built successfully, but after this commit only, the `500` error was introduced. So, maybe somewhere in these 6 commits, you can try to revert or change something to get it working. I know it will be a dirty fix, but this was all I can think of now.

![image](https://user-images.githubusercontent.com/42481964/188215019-8d74540d-9c52-4f80-a589-8043470bf20b.png)

**Comment by lvarin** [comment]
This change was done due to a very very old Kubernetes library. We were running v1, when the oldest supported version is v11. We are talking about years of differences.

So revert is not an option. I was not able to work in this since Wednesday, I hope today to be able to fix this today.

**Comment by sgalpha01** [comment]
No, reverting is not an option. I just mentioned those commits so that it will be easier for you to target the source of the error. For testing the changes I made in `cwl-wes`, I just replaced the line
```Java
Optional.ofNullable(resources).map(TesResources::getRamGb).ifPresent(ramGb -> container.getResources().putRequestsItem(RESOURCE_MEM_KEY, new QuantityFormatter().parse(ramGb.toString() + RESOURCE_MEM_UNIT)));
```
which is present here: https://github.com/elixir-cloud-aai/tesk-api/blob/master/src/main/java/uk/ac/ebi/tsc/tesk/k8s/convert/TesKubernetesConverter.java#L154
with
```Java
Optional.ofNullable(resources).map(TesResources::getRamGb).ifPresent(ramGb -> container.getResources().putRequestsItem(RESOURCE_MEM_KEY, new QuantityFormatter().parse((ramGb>0.004 ? String.format("%.3f", ramGb) : "0.004") + RESOURCE_MEM_UNIT)));
```
which is the part of legacy code. I've made this change locally, and it bypasses the error. I do realise that this is not a solution.

**Comment by lvarin** [comment]
This is very useful. The problem might be that the value generated now has too many decimals? 

I will check an hybrid of both codes and see

**Comment by lvarin** [comment]
Yes! reducing the number of decimals solved the problem.

**Comment by sgalpha01** [comment]
Addressed in #43
